File: fraud_detection/email_detection.js
async function makePrediction() {
    const feature1 = document.getElementById('emailContent').value;
 
    const features = [parseFloat(feature1)];

    const response = await fetch('http://127.0.0.1:8000/predict', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({
            features: features
        }),
    });

    const data = await response.json();

    if (response.ok) {
        document.getElementById('result').textContent = `Prediction: ${data.prediction}`;
    } else {
        document.getElementById('result').textContent = `Error: ${data.detail}`;
    }
}

File: fraud_detection/email_detection.py
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel
from backend.main_file import predict  # Import the predict function

app = FastAPI()

# Define the request body for the prediction
class PredictRequest(BaseModel):
    features: str

# Define a prediction route
@app.post("/predict")
async def get_prediction(request: PredictRequest):
    try:
        prediction = predict(request.features)
        return {"prediction": prediction}
    except Exception as e:
        raise HTTPException(status_code=400, detail=f"Error making prediction: {str(e)}")
